website/blog/testing.md

```diff
@@ -0,0 +1,512 @@
+# Lower Test Toil With Better Local Testing
+
+TODO(mwhittaker): Write an intro paragraph. Describe, briefly, the three kinds
+of testing we'll look at, and explain how they increase in complexity. Also
+maybe give a quick refresher on Service Weaver and components?
+
+## 1. Unit Testing
+
+Service Weaver has a [`weavertest` package][unit_testing] that makes it easy to
+**unit test** [components][components]. For example, consider the following
+`Adder` component:
+
+```go
+type Adder interface {
+    Add(context.Context, int, int) (int, error)
+}
+
+type adder struct {
+    weaver.Implements[Adder]
+}
+
+func (*adder) Add(_ context.Context, x, y int) (int, error) {
+    return x + y, nil
+}
+```
+
+We can use the `weavertest` package to unit test the `Adder` component using
+[idiomatic Go unit tests][go_testing]:
+
+```go
+package main
+
+import (
+    "context"
+    "testing"
+
+    "github.com/ServiceWeaver/weaver"
+    "github.com/ServiceWeaver/weaver/weavertest"
+)
+
+func TestAdd(t *testing.T) {
+     runner := weavertest.Local
+     runner.Test(t, func(t *testing.T, adder Adder) {
+         got, err := adder.Add(context.Background(), 1, 2)
+         if err != nil {
+             t.Fatal(err)
+         }
+         if want := 3; got != want {
+             t.Fatalf("got %q, want %q", got, want)
+         }
+     })
+}
+```
+
+In the code above, `weavertest.Local` is a unit test runner that runs all
+components locally. The call to `runner.Test` executes the provided unit
+test, instantiating all the necessary components (i.e. `Adder`). You can run the
+unit test above by running `go test`.
+
+## 2. Integration Testing
```

Comment:
TODO: Mention fakes.

Reply:
I would add fakes as ## 3. (because it applies both to 1 and 2), and also give a quick example on how to use it.